llvm/trunk/llvm/lib/DWARFLinker/Parallel/DWARFLinkerImpl.cpp: 3 * used uninitialised ?

On today's build of clang, gcc says:

/home/dcb38/llvm/trunk/llvm/lib/DWARFLinker/Parallel/DWARFLinkerImpl.cpp:25:40: warning: member ‘llvm::dwarf_linker::parallel::DWARFLinkerImpl::GlobalData’ is used uninitialized [-Wuninitialized]
/home/dcb38/llvm/trunk/llvm/lib/DWARFLinker/Parallel/DWARFLinkerImpl.cpp:26:27: warning: member ‘llvm::dwarf_linker::parallel::DWARFLinkerImpl::GlobalData’ is used uninitialized [-Wuninitialized]
/home/dcb38/llvm/trunk/llvm/lib/DWARFLinker/Parallel/DWARFLinkerImpl.cpp:26:55: warning: member ‘llvm::dwarf_linker::parallel::DWARFLinkerImpl::GlobalData’ is used uninitialized [-Wuninitialized]

Source code is

DWARFLinkerImpl::DWARFLinkerImpl(MessageHandlerTy ErrorHandler,
                                 MessageHandlerTy WarningHandler,
                                 TranslatorFuncTy StringsTranslator)
    : UniqueUnitID(0), DebugStrStrings(GlobalData),
      DebugLineStrStrings(GlobalData), CommonSections(GlobalData) {
  GlobalData.setTranslator(StringsTranslator);
  GlobalData.setErrorHandler(ErrorHandler);
  GlobalData.setWarningHandler(WarningHandler);
}

--- a/llvm/lib/DWARFLinker/Parallel/DWARFLinkerImpl.h
+++ b/llvm/lib/DWARFLinker/Parallel/DWARFLinkerImpl.h
@@ -351,6 +351,9 @@ protected:
   /// \defgroup Data members accessed sequentially.
   ///
   /// @{
+  /// Data global for the whole linking process.
+  LinkingGlobalData GlobalData;
+
   /// DwarfStringPoolEntries for .debug_str section.
   StringEntryToDwarfStringPoolEntryMap DebugStrStrings;
 
@@ -368,9 +371,6 @@ protected:
 
   /// Overall compile units number.
   uint64_t OverallNumberOfCU = 0;
-
-  /// Data global for the whole linking process.
-  LinkingGlobalData GlobalData;
   /// @}
 };
 

